### PR TITLE
Scripts: add a script to customize snark work production

### PR DESCRIPTION
**Pull request by anne-laure-s**
This MR adds a script & change a bit some existant code in `run_snark_worker.ml`.

The script fetch a range of work (size of the range is given by user), spawn a process and runs `run_snark_worker.ml` (perform & submit work) for each fetched work in parallel.

This script is provided as an example for users who want to customize their work production process and have more refined strategy than current selection methods.

This MR starts with commit `Scripts: add a script for fetching, performing and submitting snark work` (https://github.com/MinaProtocol/mina/commit/74241548e4957223954bf6e9e53087dbd5eb06ac)

### How to test

Run a mina node. Once the pending snark work list is filled with some works, you can run the script and see the new work being added to the snark pool.

### CODE REVIEW

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
!ci-build-me

**Comment by svv232** [comment]
I tested this by running

```python
python scripts/snark_work_strategy.py --batch-size 5 --bid 7.00 --start 1 --public-key B62qiy32p8kAKnny8ZFwoMhYpBppM1DWVCqAPBYNcXnsAHhnfAAuXgg
```

and verified that bids existed in my local snark pool 
for example:
```
      {
        "fee": "2500000",
        "prover": "B62qiy32p8kAKnny8ZFwoMhYpBppM1DWVCqAPBYNcXnsAHhnfAAuXgg",
        "workIds": [
          1027842068,
          30740314
        ]
      },
```

**Review by svv232 — APPROVED** [review]

File: scripts/snark_work_strategy.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    try:
+        # Create a temporary file to store the work_spec content which may be too large for the command line
+        with tempfile.NamedTemporaryFile(
+            delete=False, mode="w+", encoding="utf-8"
+        ) as temp_file:
+            temp_file.write(f"{work_spec}")
+            temp_file.flush()  # Ensure the content is written to disk
+            temp_file.seek(0)
+            cmd = [
+                    "_build/default/src/lib/snark_worker/standalone/run_snark_worker.exe",
```
Comment (dkijania) [inline scripts/snark_work_strategy.py:105]: I would propose to introduce const for this app and place it on top of this script with meaningful comment that this script requires run_snark_worker app being already built

**Review by dkijania — CHANGES_REQUESTED** [review]
I had one small remark

File: scripts/snark_work_strategy.py (hunk 1 above)
Comment (anne-laure-s) [reply scripts/snark_work_strategy.py]: Done

**Comment by anne-laure-s** [comment]
!ci-build-me

**Comment by svv232** [comment]
I retested this with the most recent changes, and was successful. I started a devnet node and ran 
```
python scripts/snark_work_strategy.py --batch-size 2 --bid 00696969 --start 1 --public-key B62qiy32p8kAKnny8ZFwoMhYpBppM1DWVCqAPBYNcXnsAHhnfAAuXgg
```
with this result in graphql
```graphql
query MyQuery {
  version
  snarkPool {
    fee
    prover
    workIds
  }
}
```

```json
{
  "data": {
    "version": "94ab9ad6a4349e34370adbb620ca2ba978a201b4",
    "snarkPool": [
      {
        "fee": "696969000000000",
        "prover": "B62qiy32p8kAKnny8ZFwoMhYpBppM1DWVCqAPBYNcXnsAHhnfAAuXgg",
        "workIds": [
          443021703,
          463319953
        ]
      },
      {
        "fee": "696969000000000",
        "prover": "B62qiy32p8kAKnny8ZFwoMhYpBppM1DWVCqAPBYNcXnsAHhnfAAuXgg",
        "workIds": [
          315512482,
          289622495
        ]
      }
    ]
  }
}```

**Review by dkijania — APPROVED** [review]

**Pull request merged**